app/controllers/application_controller.rb

```diff
@@ -17,7 +17,14 @@ class ApplicationController < ActionController::Base
     error_too_many_requests(exc)
   end
 
-  PUBLIC_CONTROLLERS = %w[shell async_info ga_events service_worker omniauth_callbacks registrations].freeze
+  PUBLIC_CONTROLLERS = %w[shell
+                          async_info
+                          ga_events
+                          service_worker
+                          omniauth_callbacks
+                          registrations
+                          confirmations
```

Comment:
Added confirmations and passwords as controllers that shouldn't have been in the _not allowed_ camp in the first place. Not really related to PR.